x/wasm/internal/keeper/cosmwasm/contract.go

```diff
@@ -1,42 +1,74 @@
 package cosmwasm
 
 import (
-	wasmTypes "github.com/CosmWasm/go-cosmwasm/types"
-	sdk "github.com/cosmos/cosmos-sdk/types"
+	cosmwasmv1 "github.com/CosmWasm/go-cosmwasm/types"
+	channeltypes "github.com/cosmos/cosmos-sdk/x/ibc/04-channel/types"
 )
 
+type IBCEndpoint struct {
+	Channel string `json:"channel"`
+	Port    string `json:"port"`
+}
+
+type IBCChannel struct {
+	Endpoint             IBCEndpoint
+	CounterpartyEndpoint IBCEndpoint
+	Order                channeltypes.Order
+	Version              string
+	// CounterpartyVersion can be nil when not known this context, yet
+	CounterpartyVersion *string `json:"counterparty_version,omitempty"`
+}
+
+type IBCPacket struct {
+	Data []byte
+	// identifies the channel and port on the sending chain.
+	Source IBCEndpoint
+	// identifies the channel and port on the receiving chain.
+	Destination IBCEndpoint
+	Sequence    uint64
+	// block height after which the packet times out
+	TimeoutHeight uint64
+	// block timestamp (in nanoseconds) after which the packet times out
+	TimeoutTimestamp uint64
+}
+
+type IBCAcknowledgement struct {
+	Acknowledgement []byte    `json:"acknowledgement"`
+	OriginalPacket  IBCPacket `json:"original_packet"`
+}
+
 type IBCPacketReceiveResponse struct {
 	// Acknowledgement contains the data to acknowledge the ibc packet execution
 	Acknowledgement []byte `json:"acknowledgement"`
 	// Messages comes directly from the contract and is it's request for action
-	Messages []sdk.Msg `json:"messages,omitempty"`
+	Messages []CosmosMsg `json:"messages,omitempty"`
 	// Log contains event attributes to expose over abci interface
-	Log []wasmTypes.LogAttribute `json:"log,omitempty"`
+	Log []cosmwasmv1.LogAttribute `json:"log,omitempty"`
 }
 
 type IBCPacketAcknowledgementResponse struct {
-	Messages []sdk.Msg                `json:"messages"`
-	Log      []wasmTypes.LogAttribute `json:"log"`
+	Messages []CosmosMsg               `json:"messages"`
```

Comment:
Ah, nice naming here, makes it clear which ones are the same as current, which are new... Like the new CosmosMsg type